### ISSUE TITLE
Patched isn't working in 1.19.2

### ISSUE THREAD

**Issue by Cristelknight** [EnderTurret/PatchedMod#4]
[pack.zip](https://github.com/EnderTurret/PatchedMod/files/11779404/pack.zip)
I tried to use patched to modify biomes. i could only get this pack to work in 1.20.1, but not in 1.19.2.

**Comment by EnderTurret** [comment]
The reason you only got modifying biomes working in 1.20.1 is because in 1.19.3 Mojang converted all the biomes to json files. In 1.19.2 they were still hard-coded (which is why you can't patch them.)

The only way you can patch biomes in 1.19.2 is if some other data pack or mod defines every vanilla biome in json (using say, the worldgen export). However, this would likely break any mod that does the fancy modifying-code-biomes-using-loader-provided-utilities thing (i.e., Forge biome modifiers or the Fabric biome modification API).

**Comment by Cristelknight** [comment]
Oh i read that. I got confused that it didn't show any packs when using the command I thought it would still display the packs which have the flag on in the pack.mcmeta file. I tested so long that I forgot that biome thing. How could I be so stupid xD

**Comment by Cristelknight** [comment]
i can't get it to work in 1.19.2 idk what i'm doing wrong.

**Comment by EnderTurret** [comment]
Likely you're running into some more 1.19.2 era bugs.

In particular, 1.19.2 Patched doesn't correctly handle Fabric's every-mod-in-one resource/data pack class (`GroupResourcePack`). This is because Patched asks it for the `pack.mcmeta` file and since it's a collection of many packs, it (correctly) doesn't give Patched one (because there could be many of them). Patched does correctly handle Forge's multi-pack class (`DelegatingPackResources`), but I only recently fixed the Fabric side of things (in 1.19.3).

If I remember correctly, the other major bug on 1.19.2 is that due to some funny Mojang code `.zip` resource/data packs don't allow listing files in all namespaces (something Patched tries to do). This only affects `/patched list patches [...]` (and it's been fixed in the most recent 1.20 build), so it's more of a cosmetic problem.

My guess (based on your original message) is that you're running into the first issue I've listed (Fabric mods not being handled correctly).
